builds/e2e/nested-longhaul.yaml

```diff
@@ -17,235 +17,65 @@ stages:
-          twinTester.twinUpdateFailureThreshold: '$(twinTester.twinUpdateFailureThreshold)'
-          twinTester.edgeHubRestartFailureTolerance: '$(twinTester.edgeHubRestartFailureTolerance)'
-          metricsCollector.metricsEndpointsCSV: '$(metricsCollector.metricsEndpointsCSV)'
-          metricsCollector.scrapeFrequencyInSecs: '$(metricsCollector.scrapeFrequencyInSecs)'
-          metricsCollector.uploadTarget: '$(metricsCollector.uploadTarget)'
-          metricsCollector.hostPlatform: '$(hostPlatform)'
-          longHaul.parentHostname: '$(parentName)'
-          longHaul.parentEdgeDevice: '$(parentDeviceId)'
-          testResultCoordinator.storageAccountConnectionString: '$(EdgeLonghaulStorageAccountConnString)'
-          quickstart.package.name: '$(quickstart.package.name)'
-          aziotis.artifact.name: '$(aziotis.artifact.name)'
-          aziotis.package.filter: '$(aziotis.package.filter)'
+    parameters:
+      transportType: amqp
+  - template: templates/nested-longhaul-deploy-amd64.yaml
+    parameters:
+      transportType: amqp
+  - template: templates/nested-parent-vm-setup.yaml
+    parameters:
+      transportType: mqtt
+  - template: templates/nested-longhaul-deploy-amd64.yaml
+    parameters:
+      transportType: mqtt
+
   - job:  Clean_up      
     dependsOn:
-      - SetupVM_level5 
-      - SetupVM_level4
-      - Deploy_Longhaul_Linux_Amd64
-      - Deploy_Longhaul_Linux_Arm32
-      - Deploy_Longhaul_Linux_Arm64
+      - SetupVM_level5_amqp
+      - SetupVM_level5_mqtt
+      - SetupVM_level4_amqp
+      - SetupVM_level4_mqtt
+      - Deploy_Longhaul_Linux_Amd64_amqp
+      - Deploy_Longhaul_Linux_Amd64_mqtt
     condition: eq(variables['test.pipeline'], 'true')
     displayName: Clean up identities and unlock agents
     variables:    
-      deviceLvl5AgentName: $[ dependencies.SetupVM_level5.outputs['lock_test_agent.agentName'] ]         
-      deviceLvl4AgentName: $[ dependencies.SetupVM_level4.outputs['lock_test_agent.agentName'] ]       
-      amd64AgentName: $[ dependencies.Deploy_Longhaul_Linux_Amd64.outputs['lock_test_agent.agentName'] ]  
-      arm32AgentName: $[ dependencies.Deploy_Longhaul_Linux_Arm32.outputs['lock_test_agent.agentName'] ]  
-      arm64AgentName: $[ dependencies.Deploy_Longhaul_Linux_Arm64.outputs['lock_test_agent.agentName'] ]  
+      deviceLvl5AmqpAgentName: $[ dependencies.SetupVM_level5_amqp.outputs['lock_test_agent.agentName'] ]
+      deviceLvl5MqttAgentName: $[ dependencies.SetupVM_level5_mqtt.outputs['lock_test_agent.agentName'] ]
+      deviceLvl4AmqpAgentName: $[ dependencies.SetupVM_level4_amqp.outputs['lock_test_agent.agentName'] ]
+      deviceLvl4MqttAgentName: $[ dependencies.SetupVM_level4_mqtt.outputs['lock_test_agent.agentName'] ]
+      amd64AmqpAgentName: $[ dependencies.Deploy_Longhaul_Linux_Amd64_amqp.outputs['lock_test_agent.agentName'] ]
+      amd64MqttAgentName: $[ dependencies.Deploy_Longhaul_Linux_Amd64_mqtt.outputs['lock_test_agent.agentName'] ]
```

Comment:
You technically can use a matrix strategy to parallelize the run time and shrink the code base of the clean up process. It would require more work to do so... 
I'm gonna leave this here. You can make a call if you want to implement it or not.
https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#matrix

Reply:
Discussed this offline - this would also require a large refactor. A lot of the jobs have dependencies that don't work quite well with matrix